### PR TITLE
Async support

### LINKED ISSUE #103: Search lyrics asynchronously
check [gidle_future](https://docs.rs/gidle_future/latest/gidle_future/#tokio)
[JoinSet](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html)

### PR DESCRIPTION
**Pull request by poly000**
fix #103

### CODE REVIEW

File: src/app/window/imp.rs [hunk 1]
```diff
@@ -107,14 +107,14 @@ impl WidgetImpl for Window {
...
+    fn close_request(&self) -> Propagation {
         // Save window size
         self.obj()
             .save_window_state()
             .expect("Failed to save window state");
 
-        // Don't inhibit the default handler
-        Inhibit(false)
+        // Don't invoke the default handler
+        Propagation::Stop
```
Comment (taoky) [inline src/app/window/imp.rs:117]: https://gtk-rs.org/blog/2023/08/28/new-release.html#replacement-of-glibcontinue--glibinhibit

```
glib::Inhibit(false) -> glib::Propagation::Proceed
```

File: src/lib.rs [hunk 2]
```diff
@@ -17,13 +22,15 @@ pub mod utils;
 pub const APP_ID: &str = "io.poly000.waylyrics";
 
 thread_local! {
-    pub static CONFIG_HOME: RefCell<String> = RefCell::new(String::new());
     pub static CACHE_DIR: RefCell<String> = RefCell::new(String::new());
-    pub static THEME_PATH: RefCell<PathBuf> = RefCell::new(PathBuf::new());
+        pub static THEME_PATH: RefCell<PathBuf> = RefCell::new(PathBuf::new());
```
Comment (taoky) [inline src/lib.rs:26]:
```suggestion
    pub static THEME_PATH: RefCell<PathBuf> = RefCell::new(PathBuf::new());
```

File: src/main.rs [hunk 3]
```diff
@@ -130,12 +133,17 @@ fn build_ui(app: &Application) -> Result<()> {
 
     if let Some(base_url) = qqmusic_api_base_url {
         let base_url = url::Url::from_str(&base_url)?;
-        QQMUSIC_API_CLIENT.set(Some(QQMusicApi::new(base_url)));
+        let _ = QQMUSIC_API_CLIENT.set(Some(QQMusicApi::new(base_url)));
```
Comment (taoky) [inline src/main.rs:136]: Is `let _ = ` necessary here?

Comment (poly000) [reply src/main.rs:136]: No, it's here to mute compiler warning about unused result, but that's what we expect

**Pull request merged**